xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/FormContainerElement.java

```diff
@@ -73,20 +73,25 @@ protected WebElement getFormElement()
 
     public void fillFieldsByName(Map<String, String> valuesByNames)
     {
-        Map<WebElement, String> valuesByElements = new LinkedHashMap<>((int) (valuesByNames.size() / 0.75));
-
+        Map<WebElement, String> valuesByElements = new LinkedHashMap<>();
+        
+        WebElement lastElement = null;
         for (String name : valuesByNames.keySet()) {
-            valuesByElements.put(getFormElement().findElement(By.name(name)), valuesByNames.get(name));
+            lastElement = getFormElement().findElement(By.name(name));
+            valuesByElements.put(lastElement, valuesByNames.get(name));
         }
         fillFieldsByElements(valuesByElements);
         
         /* Register password confirmation is usually the last element that needs to be validated by liveValidation.
           This wait allows to solve a race condition between the form submission and the computation of the status of
           those fields. We force the status to be solved before we try anything else, especially submitting the form.
+          Unfortunately in Java17 we do not have lastEntry() from LinkedHashMaps, 
+          so we use a few non optimized operations instead. 
+          This is okay because the Map should not contain a lot of elements.
           */
-        if (valuesByNames.containsKey("register2_password")) {
-            getDriver().waitUntilCondition(driver -> !getFormElement().findElement(By.name("register2_password"))
-                .getAttribute(CLASS_ATTRIBUTE).isEmpty());
+        if(!valuesByElements.isEmpty()) {
```

Comment:
now you could also check if lastElement != null :)

Reply:
Technically if the list is not empty the lastElement should be properly set at least once because we do at least one pass in the loop above . It doesn't hurt to check though . Added in https://github.com/xwiki/xwiki-platform/pull/3675/commits/3efdb2ac94adcd375e4440b538e7a8c4f0405c2e :+1: 
I didn't run tests again, I suppose this change won't break things.

Reply:
Yeah sorry my "also" was confusing, I meant you could have check only for the lastElement != null, would be more accurate given what you're doing inside. But nevermind I'll merge it like that, I've been annoying enough on this one ;)